docs/architecture/api-packages.md

```diff
@@ -0,0 +1,159 @@
+      "type": "object",
+      "required": ["events", "apis"],
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Description"
+        },
+        "metadata": {
+          "type": "object",
+          "description": "Additional Metadata for Service Class representation"
+        },
+        "events": {
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "pattern": "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+In a result, the `api-packages` label value is validated. Also, the API Package name uniqueness within Application is guaranteed. Event and API Definition existence is not checked, apart from verifying the UUID type of the ID arrays.
+
+This schema cannot be modified by user. We validate this JSON schema against internal JSON schema, similarly to `scenarios` LabelDefinition.
+
+### Assumptions
+
+- API or Event Definition can be a part of API Package only within the same Application entity.
+- API Package name doesn't have to be unique within tenant.
+- One API or Event Definition can belong to only one API Package. If an API or Event Definition is assigned to multiple API packages, or some items with provided IDs doesn't exist, then Runtime Agent reports a configuration sync error.
+
+### Pros
+
+- No breaking changes in our API
+- Consistent implementation (identical to Scenarios)
+- Static label validation
+
+### Cons
+
+- Additional logic on client side (see the [Client Usage](#client-usage) section)
+- _(not important)_ No validation for existence of API/Event Definition IDs
+
+## Example
+
+> **NOTE:** For clarity, in this example IDs has been changed from UUID type to string.
+
+1. User creates Application `sample-app`.
+1. User registers three different API definitions with the following IDs: `foo`, `bar` and `baz`.
+1. In order to create API Package called `sample-package`, User sets the following label for `sample-app` Application:
+
+   **Key**: `api-packages`
+   **Value:**
+
+   ```json
+   {
+     "sample-package": {
+       "description": "A Sample API Package",
+       "metadata": {
+         "displayName": "Sample Package",
+         "documentationURL": "",
+         "supportURL": ""
+       },
+       "events": [],
+       "apis": ["bar", "baz"]
+     }
+   }
+   ```
+
+1. Runtime Agent configures Runtime.
+1. In Service Catalog, there are two Service Classes: `foo` and `sample-package`, which contains single Service Plan with `bar` and `baz` APIs grouped.
+
+## Client Usage
+
+The following section describes client usage of the API Packages.
+
+### Runtime Agent
+
+To create Service Classes in Kyma Service Catalog from Application APIs and Event Definitions, Runtime Agent (or another component on Runtime side) has to populate the Service Class list.
+
+In order to do so, Runtime Agent executes the following steps:
+
+1. Query Applications for Runtime
```

Comment:
If we have many agents, every agent has to implement the same logic. Why Compass cannot prepare "ready-to-use" data and store it in another label?

Reply:
As discussed, in future we will think about doing this work on Director by extending our API.